vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_util.cc

```diff
@@ -319,16 +312,43 @@ type::ExternalWalletPtr ResetWallet(type::ExternalWalletPtr wallet) {
   wallet = type::ExternalWallet::New();
   wallet->type = constant::kWalletUphold;
 
-  if (status != type::WalletStatus::NOT_CONNECTED) {
-    if (status == type::WalletStatus::VERIFIED) {
-      wallet->status = type::WalletStatus::DISCONNECTED_VERIFIED;
-    } else {
-      wallet->status = type::WalletStatus::DISCONNECTED_NOT_VERIFIED;
-    }
+  if (status == type::WalletStatus::VERIFIED) {
+    wallet->status = type::WalletStatus::DISCONNECTED_VERIFIED;
   }
 
   return wallet;
 }
 
+template <typename T, typename... Ts>
+bool one_of(T&& t, Ts&&... ts) {
+  bool match{};
+
+  static_cast<void>(std::initializer_list<bool>{
+      (match = match || std::forward<T>(t) == std::forward<Ts>(ts))...});
+
+  return match;
+}
+
+void LogWalletStatusChange(LedgerImpl* ledger,
+                           base::Optional<type::WalletStatus> from,
+                           type::WalletStatus to) {
+  DCHECK(ledger);
+  DCHECK(!from ||
+         one_of(*from, type::WalletStatus::NOT_CONNECTED,
+                type::WalletStatus::DISCONNECTED_VERIFIED,
+                type::WalletStatus::PENDING, type::WalletStatus::VERIFIED));
+  DCHECK(one_of(to, type::WalletStatus::NOT_CONNECTED,
+                type::WalletStatus::DISCONNECTED_VERIFIED,
+                type::WalletStatus::PENDING, type::WalletStatus::VERIFIED));
+
+  std::ostringstream oss{};
+  if (from) {
+    oss << *from << ' ';
+  }
+  oss << "==> " << to;
```

Comment:
You might like `base::StringPrintf` for this kind of thing.

Reply:
Yeah, I would.🙂
`WalletStatusToString()` is internal to `ledger.mojom-shared.cc`, only `operator<<()` is exported, so I had to use a `std::basic_ostream<>` descendant here.

Reply:
Ah OK, I didn't notice it's a mojo type, fun...!